hydra-node/src/Hydra/Network.hs

```diff
@@ -39,13 +39,20 @@ import Text.Show (Show (show))
 -- * Hydra network interface
 
 -- | Handle to interface with the hydra network and send messages "off chain".
-newtype Network m msg = Network
+data Network m msg = Network
   { -- | Send a `msg` to the whole hydra network.
     broadcast :: msg -> m ()
+    -- | Returns the list of connected peers in the hydra network.
+  , getPeers :: m [Host]
+    -- | Modify the list of peers in the hydra network,
+    -- causing the network to reinitialize and reconnect to the new list of peers.
+  , setPeers :: [Host] -> m ()
   }
 
 instance Contravariant (Network m) where
-  contramap f (Network bcast) = Network $ \msg -> bcast (f msg)
+  contramap f (Network bcast getPeers setPeers) = Network bcast' getPeers setPeers
+   where
+    bcast' = bcast . f
```

Comment:
Two small notes: 

- I don't think the let-binding on `bcast'` buys us anything here; the name isn't any-more significant and there's no type signature anyway. 
- It's only ever needed to change the `broadcast` field to implement `contramap` here, so rather than bringing everyone in scope, I'd leverage NamedFieldPuns' notation and only modify the `broadcast` field:

```hs
contramap f network@Network{broadcast} = network{broadcast = broadcast . f}
```

This way, if new fields are added, or fields are changed, one needs not to come back and change this definition.